superset/common/query_context_processor.py

```diff
@@ -455,6 +455,13 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
                 for metric in metric_names
             }
 
+            # When the original query has limit or offset we wont apply those
```

Comment:
Wow this method has gotten large! Thinking about refactors here and factoring out a `processing_time_offset` (a method for each time offset) might help a little in reading this, but probably breaking down further would be good too. 

Doing the refactor in this PR as part of this PR would make it harder to spot the diff, so I'll approve this one, but hoping we can refactor some of these into more bite-sized methods. Maybe a goal could be to remove the `disable=too-many-locals,too-many-statements` above.